Release helper script

saulo-p opened the pull request.

saulo-p committed:
deployment helper: initial commit

script that assists release creation for the repositories within the
sct-data github organization

saulo-p committed:
create_release: create a release to the target repository

bases the release on a given tag

saulo-p committed:
upload_github_asset: uploads a local file to a target release

saulo-p committed:
download default release asset function

given a release id, download the default zipball associated to it

saulo-p committed:
upload to osf: first iteration with subprocess spawning

saulo-p committed:
osf upload: implement with osfclient api instead of subprocess

also, returns the download link

saulo-p committed:
arg parsing

zougloub commented on deployment-helper.py:
```diff
@@ -0,0 +1,170 @@
+	logger.info("Uploading release asset")
+
+	url = "https://uploads.github.com/repos/sct-data/{}/releases/{}/assets?name={}" \
+		.format(target_repo, release_id, asset_path)
```
Basename

zougloub commented on deployment-helper.py:
```diff
@@ -0,0 +1,170 @@
+
+	def parse_args():
+		parser = argparse.ArgumentParser(
+		description="Create a release on one of the repositories within sct-data organization")
```
Since this script would be used, we should use the SCT coding standard.
I think you can use `black` to reformat this script.

saulo-p replied:
Sure. Where do you think this script will sit? Maybe `sct/dev`?

zougloub commented on deployment-helper.py:
```diff
@@ -0,0 +1,170 @@
+
+if __name__ == "__main__":
+	res = main()
+	raise SystemExit()
```
unless you `raise SystemExit(res)` the statement is useless

zougloub commented on deployment-helper.py:
```diff
@@ -0,0 +1,170 @@
+		return args
+
+
+	return
```
Useless statement

saulo-p committed:
validate repository and tag at argparsing

validate if the repository exists within organization and if tag exists
within repository.

saulo-p commented on deployment-helper.py:
```diff
@@ -0,0 +1,207 @@
+	}
+
+	with io.open(asset_path, "rb") as fin:
+		payload = fin.read()
```
@zougloub, do you think this line could be a problem for large files?

saulo-p committed:
update release: include osf download url to release body

saulo-p committed:
download asset: return asset path

saulo-p committed:
main script

also, adds misc fixes/improvements
